zanata-war/src/main/java/org/zanata/adapter/GenericPropertiesAdapter.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2015, Red Hat, Inc. and individual contributors as indicated by the
+ * @author tags. See the copyright.txt file in the distribution for a full
+ * listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This software is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA, or see the FSF
+ * site: http://www.fsf.org.
+ */
+
+package org.zanata.adapter;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
+import org.zanata.adapter.properties.PropReader;
+import org.zanata.adapter.properties.PropWriter;
+import org.zanata.common.ContentState;
+import org.zanata.common.LocaleId;
+import org.zanata.exception.FileFormatAdapterException;
+import org.zanata.rest.dto.resource.Resource;
+import org.zanata.rest.dto.resource.TranslationsResource;
+import org.zanata.util.FileUtil;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+
+/**
+ * Properties file adapter for read and write.
+ *
+ * See
+ * {@link org.zanata.adapter.PropertiesLatinOneAdapter}
+ * {@link org.zanata.adapter.PropertiesUTF8Adapter}
+ *
+ * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
+ */
+@Slf4j
+public class GenericPropertiesAdapter implements FileFormatAdapter {
+
+    public static final String ISO_8859_1 = Charsets.ISO_8859_1.name();
+
+    public static final String UTF_8 = Charsets.UTF_8.name();
+
+    private final String charset;
+
+    public GenericPropertiesAdapter(String charset) {
+        // check that charset is valid, throw exception if not
```

Comment:
Easy fix for this: change `charset` to type `PropWriter.CHARSET`, then in this constructor do `this.charset = toPropWriterCharset(charset);`.

That will do the check, and it means the code below can use `charset` directly.